crates/core/tedge_mapper/src/main.rs

```diff
@@ -115,7 +128,97 @@ async fn main() -> anyhow::Result<()> {
         component.init(&mapper_opt.config_dir).await
     } else if mapper_opt.clear {
         component.clear_session().await
+    } else if component.session_name().eq("tedge-mapper-az") {
+        run_mapper(component.session_name(), config).await
+    } else if component.session_name().eq("tedge-mapper-aws") {
+        run_mapper(component.session_name(), config).await
     } else {
         component.start(config, &mapper_opt.config_dir).await
     }
 }
+
+async fn run_mapper(mapper_name: &str, config: TEdgeConfig) -> Result<(), anyhow::Error> {
+    let clock = Box::new(WallClock);
+    match mapper_name {
+        "tedge-mapper-aws" => {
+            let (mut runtime, mut mqtt_actor) = start_basic_actors(mapper_name, &config).await?;
+
+            let aws_converter = AwsConverter::new(
+                config.query(AwsMapperTimestamp)?.is_set(),
+                clock,
+                SizeThreshold(255 * 1024),
+            );
+            let mut aws_converting_actor = ConvertingActor::builder("AwsConverter", aws_converter);
+            mqtt_actor.register_peer(
+                AwsConverter::in_topic_filter(),
+                aws_converting_actor.get_sender(),
+            );
+            aws_converting_actor.register_peer(NoConfig, mqtt_actor.get_sender());
+
+            runtime.spawn(aws_converting_actor).await?;
+            runtime.spawn(mqtt_actor).await?;
+            runtime.run_to_completion().await?;
+        }
```

Comment:
It would be good to move this code in the AWS mapper start method.

Reply:
fixed